client/state/user-settings/reducer.js

```diff
@@ -42,7 +44,19 @@ export const unsavedSettings = ( state = {}, action ) => {
 	}
 };
 
+export const pendingPasswordChange = ( state = false, action ) => {
+	switch ( action.type ) {
+		case USER_SETTINGS_SAVE:
+		case USER_SETTINGS_UPDATE:
+		case USER_SETTINGS_UPDATE_FAILURE: {
+			return action.type === USER_SETTINGS_SAVE && Boolean( action.settingsOverride?.password );
```

Comment:
This `type` check feels odd. Should we just do whatever is expected under each of the action types? Seems more straightforward and common to me.

Reply:
addressed in fa571bd64379d9fcb49825e2f5a3a3f7edbad067